index.js

```diff
@@ -36,20 +37,30 @@ export const useRouter = () => {
   return globalRef.v || (globalRef.v = buildRouter());
 };
 
+// if you would like to customize, for example:
+//   - to manage state or document.title
+// override useNavigate through "Router" & don't use this default "useNavigate" Hook
+export const useNavigate = () => {
+  const { base } = useRouter();
+  return useCallback((to, { replace } = {}) =>
+    history[replace ? eventReplaceState : eventPushState](null, "", base + to), [base]);
```

Comment:
This will mess up the separation of concerns: wouter itself should know nothing about the implementation details of the location interface. It'll break existing apps as well: those who use custom locations hooks, for example hash-based or internal (global state) routing.